Add missing API functions and doxygen style documentation for C bindings

BenjaminRueth opened the pull request:
this PR resulted from splitting up https://github.com/precice/precice/pull/398

IshaanDesai commented:
After a discussion with @BenjaminRueth I think it is a good idea to update the documentation of the C Bindings along with adding the missing API functions

BenjaminRueth commented:
> I thought we remove `src/precice/bindings/c/Constant*` already. I am confused. We should keep it consistent to the C++ API.

In the current version on develop this is also the case. We have to merge develop into this branch before we can merge back.

BenjaminRueth commented:
I merged develop into this branch. I just realized that `precicec_actionWriteInitialData` was missing in the C bindings and added this action. This was already the case when we removed the constants in https://github.com/precice/precice/commit/945eed22727b1fdafe9051c92684a7c6752e26bb#diff-5cd53866994afe4e9816951e94a0077f.

@uekerman do you know whether this action is missing intentionally or was it just forgotten?

IshaanDesai commented on src/precice/bindings/c/SolverInterfaceC.cpp:
```diff
@@ -104,6 +122,13 @@ void precicec_fulfilledAction ( const char* action )
+{
+  assertion ( impl != nullptr );
+  std::string stringMeshName (meshName);
+  return impl->hasMesh (stringMeshName);
```
Yes you are right, resolved in the forthcoming commit.

uekerman commented:
@BenjaminRueth Was probably never needed. Nothing to worry about.

BenjaminRueth commented:
@uekerman Is everything fine now from your point of view?
